src/test/java/com/wikia/webdriver/elements/mercury/pages/SearchResultsPage.java

```diff
@@ -76,14 +85,45 @@ public boolean isNoResultsPagePresent() {
     }
   }
 
+  public boolean isLoadMoreButtonVisible() {
+    return isElementVisible(loadMoreButton);
+  }
+
   public boolean areResultsPresent() {
     try {
-      wait.forElementClickable(By.cssSelector(searchResultClass), 0);
+      wait.forElementClickable(By.cssSelector(SEARCH_RESULT_SELECTOR), 0);
       return true;
     } catch (NoSuchElementException e) {
       return false;
     } catch (TimeoutException e) {
       return false;
     }
   }
+
+  public SearchResultsPage clickLoadMoreButton() {
+    PageObjectLogging.logInfo("Click Load More button ");
+    wait.forElementClickable(loadMoreButton);
+    loadMoreButton.click();
+
+    return this;
+  }
+
+  public SearchResultsPage waitForResultsLoaded() {
+    wait.forElementNotVisible(By.cssSelector(SPINNER_SELECTOR));
+    return this;
+  }
+
+  public int getResultCardsNumber() {
+    waitForResultsLoaded();
+    return resultCards.size();
+  }
+
+  private boolean isElementVisible(WebElement element) {
+    try {
+      return element.isDisplayed();
+    } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)